src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/DbConnectionStringCommon.cs

```diff
@@ -1147,20 +962,23 @@ internal static SqlConnectionIPAddressPreference ConvertToIPAddressPreference(st
             }
         }
         #endregion
-
-        internal static bool IsValidCertificateValue(string value)
-        {
-            return string.IsNullOrEmpty(value)
-                || value.StartsWith("subject:", StringComparison.OrdinalIgnoreCase)
-                || value.StartsWith("sha1:", StringComparison.OrdinalIgnoreCase);
-        }
     }
 
     internal static class DbConnectionStringDefaults
     {
-        // all
-        //        internal const string NamedConnection           = "";
         private const string _emptyString = "";
```

Comment:
`_emptyString` is pointless. Because the runtime uses string interning all instances of the empty string will refer to the same instance and the only way to change the value of that instance is using unsafe code, if someone does this they should expect a crash and we should not attempt to protect against it. We can just use the literal `""` or `string.Empty` where we need an empty string. There is a similar constant in the ADP class which will eventually be removed.

Reply:
The `const` is a compile-time value, not runtime! Using a common constant to declare the string empty would prevent possible miss-assignment. I don't think using `_emptyString` had any drawback here.
Using `const` instead of `readonly` would be a quick solution for ADP.

Reply:
Using either a const or readonly doesn't matter, there is only one empty string unless you Create one and no-one should do that. 

Using all the extra names for "" obscures the value. If you see `DbConnectionStringDefaults.Driver` you'll think it has some special meaning, it does not, it's just "". It's harmful to readability, it's no harmful at compile or runtime.

Reply:
I replaced the `_emptyString` with `""`.
Regarding the argument about const string default values and empty string, I agree it'd obscure the value but consider it has some benefits like having reference to all usages in the one place, which means code maintenance would be much easier and safer.